packages/lib/src/components/Card/components/ClickToPay/components/CtPOneTimePasswordInput/CtPOneTimePasswordInput.tsx

```diff
@@ -0,0 +1,58 @@
+import { h } from 'preact';
+import { useCallback, useEffect, useImperativeHandle } from 'preact/hooks';
+import { forwardRef } from 'preact/compat';
+import { otpValidationRules } from './validate';
+import useCoreContext from '../../../../../../core/Context/useCoreContext';
+import useForm from '../../../../../../utils/useForm';
+import Field from '../../../../../internal/FormFields/Field';
+import renderFormField from '../../../../../internal/FormFields';
+import './CtPOneTimePasswordInput.scss';
+
+interface Props {
+    disabled: boolean;
+    errorCode?: string;
+    onChange({ data: CtPOneTimePasswordInputDataState, valid, errors, isValid: boolean }): void;
+}
+
+interface CtPOneTimePasswordInputDataState {
+    otp?: string;
+}
+
+export type CtPOneTimePasswordInputHandlers = {
+    validateInput(): void;
+};
+
+const CtPOneTimePasswordInput = forwardRef<CtPOneTimePasswordInputHandlers, Props>((props, ref) => {
+    const { i18n } = useCoreContext();
+    const formSchema = ['otp'];
+    const { handleChangeFor, data, triggerValidation, valid, errors, isValid } = useForm<CtPOneTimePasswordInputDataState>({
+        schema: formSchema,
+        rules: otpValidationRules
+    });
+
+    const validateInput = useCallback(() => {
+        triggerValidation();
+    }, [triggerValidation]);
+
+    useImperativeHandle(ref, () => ({ validateInput }));
+
+    useEffect(() => {
+        props.onChange({ data, valid, errors, isValid });
+    }, [data, valid, errors]);
+
+    return (
+        <Field label={i18n.get('One time code')} errorMessage={props.errorCode || !!errors.otp} classNameModifiers={['otp']}>
```

Comment:
Fields like this should have a `name` prop e.g. `name={'oneTimeCode}`
This will ensure that the generated `label` then has a `for` attribute that will link to a unique `id `generated on the enclosed text field

Reply:
I will add on the follow up PR 👍

Reply:
ok